### ISSUE TITLE
Todos

### ISSUE THREAD

**Issue by BasicWolf** [BasicWolf/hexagonal-architecture-django#32]
### A list of todos, too small to have an issue about them :)

- [x] ~~Domain entities should be comparable by ID only. How about creating a base class?~~
- [x] Review VotingUserRepository unit tests - hide unnecessary data to fixtures

**Comment by rotate2050** [comment]
@BasicWolf Thanks for putting this project together. I want to help you out for the second task(to making unit testing better) if you are OK. Let me know if you have specific taught on it.

**Comment by BasicWolf** [comment]
Hei @rotate2050, thank you for reaching out! Glad that it all makes sense for someone else but me. I'd love to hear your ideas!
I've recently cleaned up some of the tests, trying to hide the unnecessary fixture instantiation details under `@pytest.fixtures()`. I hope the code looks a little bit better now, but there is always a room for improvement. Please have a look at the latest code updates. 

One particular thing that I've been skipping for a while - handling UUIDs in fixtures. 
There are alternatives, like:
```
* ArticleId(UUID('06aee517-0000-0000-0000-000000000000'))
* ArticleId(UUID(int=1))
* article_id(1) # calls a function which returns the above
```
Any good thoughts about this one?

**Comment by rotate2050** [comment]
@BasicWolf Thanks for your quick response and for initiating this project. Before I talk further about my idea, I wanted to share my background a bit. 

I have worked with PHP for most of my career. I have used a hexagonal architectural pattern with PHP and I am a huge fan of it. Recently I changed my project and started working on Python Django framework. I was quite amazed that there were very few posts that talked about using the hexagonal pattern in Python Django framework. I was very happy after seeing your examples. I learned a lot. 

I am also a huge fan of the TDD framework. Tests are a great way of learning the software. So I was thinking of first going through the tests and thought about very simple improvements that you already have thought of. I would like to start with your suggestion first. Once I get more involved in it, I may come up with more suggestions. 

I liked your idea of separating out the test fixtures, that's a great place to start. As I said, I am new to Python. So I may need to spend some time on `@pytest.fixtures()`. Would it be OK if I take a week to go through it?

**Comment by BasicWolf** [comment]
Sure, feel free to contribute at any time. There are no estimates on this project :D

I learned about Hexagonal architecture and DDD in context of professional Kotlin and Spring -based application development. That was a fantastic architectural experience, the one I haven't witnessed in typical Django/Python world yet. And that is how this project was born. 

The original goal was to keep it simple stupid, but then I started over engineering things. Hopefully It got back on right track :) That is thanks to the tests. It has gone through numerous refactorings and only the test suite allowed to be certain, that behavior is not broken. 

Lately I've been trying to express the behaviour even through the unit tests. Hiding fixtures behind human-readable and behaviour-driven names is part of this. So the UUIDs problem might solve itself, since the actual values would be hidden behind fixtures and not contaminate the tests anymore.
